polyply/src/simple_seq_parsers.py

```diff
@@ -0,0 +1,236 @@
+
+    Parameters
+    ----------
+    lines: `abc.iteratable`
+        list of strings matching one letter code DNA, RNA, or AAs
+    DNA: bool
+        if the sequence matches DNA
+    RNA: bool
+        if the sequence matches RNA
+
+    Returns
+    -------
+    `:class:nx.Graph`
+        the plain residue graph matching the sequence
+    """
+    monomers = []
+    for line in lines:
+        for token in line.strip():
+            if token in ONE_LETTER_DNA and DNA:
+                resname = ONE_LETTER_DNA[token]
+            elif token in ONE_LETTER_RNA and RNA:
+                resname = ONE_LETTER_RNA[token]
+            else:
+                msg = f"Cannot find one letter residue match for {token}"
+                raise IOError(msg)
+
+            monomers.append(resname)
+
+    # make sure to set the defaults for the DNA terminals
+    if DNA:
+        monomers[0] = monomers[0] + "5"
+        monomers[-1] = monomers[-1] + "3"
+
+    seq_graph =  _monomers_to_linear_nx_graph(monomers)
+    return seq_graph
+
+def _identify_nucleotypes(comments):
+    """
+    From a comment found in the ig or fasta file, identify if
+    the sequence is RNA or DNA sequence by checking if these
+    keywords are in the comment lines. Raise an error if
+    none or conflicting information are found.
+
+    Parameters
+    ----------
+    comments: abc.itertable
+        list of comment lines
+
+    Returns
+    -------
+    bool, bool
+        is it DNA, RNA
+
+    Raises
+    ------
+    IOError
+        neither RNA nor DNA keywords are found
+        both RNA and DNA are found
+    """
+    RNA = False
+    DNA = False
+    for comment in comments:
+        if "DNA" in comment:
+            DNA = True
+
+        if "RNA" in comment:
+            print("go here")
+            RNA = True
+
+    if RNA and DNA:
+        raise IOError("Found both RNA and DNA keyword in comment. Choose one.")
+
+    if not RNA and not DNA:
+        raise IOError("Cannot identify if sequence is RNA or DNA from comment.")
+
+    return DNA, RNA
+
+def parse_ig(filehandle):
+    """
+    Read ig sequence in DNA/AA formatted file. See following link
+    for format:
+
+    https://www.animalgenome.org/bioinfo/resources/manuals/seqformats
+
+    Parameters
+    ----------
+    filehandle: :class:`pathlib.Path`
+    """
+    with open(filehandle) as file_:
+        lines = file_.readlines()
+
+    clean_lines = []
+    comments = []
+    idx = 0
+    for idx, line in enumerate(lines):
+        clean_line, comment = split_comments(line)
+        comments.append(comment)
+        if clean_line:
+            if clean_line[-1] == '1' or clean_line[-1] == '2':
+                ter_char = clean_line[-1]
+                clean_line = clean_line[:-1]
+                clean_lines.append(clean_line)
+                DNA, RNA = _identify_nucleotypes(comments)
+                seq_graph = parse_plain(clean_lines[1:], DNA=DNA, RNA=RNA)
```

Comment:
Because according to the format specification the first non-comment line will be the name and then comes the actual sequence. Probably should add a comment here

Reply:
I did not realise it was a list of lines and not a list of characters. My bad.